src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.Unix.cs

```diff
@@ -53,7 +54,9 @@ public abstract partial class TarEntry
             // If the permissions weren't set at all, don't write the file's permissions.
             if (permissions != 0)
             {
-                Interop.CheckIo(Interop.Sys.FChMod(handle, permissions), destinationFileName);
+#pragma warning disable CA1416 // Validate platform compatibility
```

Comment:
I've added this to avoid the error for using an API that is not supported on Windows.
The file is only used on Unix, so maybe there is a better way to handle this?

Reply:
I added it by mistake to `TarEntry.Unix.cs`. It is only required for `ZipFileExtensions.ZipArchiveEntry.Extract.Unix.cs`.
And there it makes sense because the file is included in `$(NetCoreAppCurrent)` target.

```
/home/tmds/repos/runtime/src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.Unix.cs(26,17): error CA1416: This call site is reachable on all platforms. 'File.SetUnixFileMode(SafeFileHandle, UnixFileMode)' is unsupported on: 'windows'. [/home/tmds/repos/runtime/src/libraries/System.IO.Compression.ZipFile/src/System.IO.Compression.ZipFile.csproj]
```
The csproj file has:
https://github.com/dotnet/runtime/blob/3282677b45e872ba97746f07914f3eb882a1d6d3/src/libraries/System.IO.Compression.ZipFile/src/System.IO.Compression.ZipFile.csproj#L4
and
https://github.com/dotnet/runtime/blob/3282677b45e872ba97746f07914f3eb882a1d6d3/src/libraries/System.IO.Compression.ZipFile/src/System.IO.Compression.ZipFile.csproj#L25-L27

Reply:
The TFMs on this project seem incorrect to me. It has changed quite a few times in the last year.

1. When I made it respect file modes in #55531 to have a separate "Unix" build
2. @ViktorHofer changed it to have an "agnostic" TFM (basically the "Windows" build) in #64518
3. Lastly in April with #67332 to add back a "Windows" TFM and make the "unix" side the agnostic build.

So unfortunately, because of these TFM changes, the "Unix" build doesn't get the attribute that says "this isn't Windows", which means the analyzer is handicapped here.

Instead of disabling the warning, I think just adding `Debug.Assert(!OperatingSystem.Windows);` at the top of the method would be best for now. Long term, I hope we can settle on a manageable TFM plan.

Reply:
> Long term, I hope we can settle on a manageable TFM plan.

@eerhardt can you please elaborate why you think that the current strategy isn't manageable? cc @ericstj

Reply:
> @buyaa-n @ViktorHofer the File.SetUnixFileMode method is marked as [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("windows")] and this particular C# file is compiled only for Unix:

[The SDK unconditionally adds supported platforms](https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedPlatforms.props#L16-L18) and one of those is Windows. The SDK itself doesn't make it possible to only target Windows, macOS or Linux via a target platform, but we in dotnet/runtime allow that via the Microsoft.DotNet.TargetFramework.Sdk.

Because of that, we likely need to remove these unconditionally set `SupportedPlatform` items and instead add them similar to how it is set for Browser conditionally.

Unrelated but also something that I noticed:

[Here](https://github.com/dotnet/runtime/blob/b1221494186e7b8e5199d892dae76481a52f5935/eng/versioning.targets#L79) you can see that the `SupportedPlatform` metadata is added for Browser when the platform is Browser, or when it's agnostic and the library doesn't have a Browser specific implementation.

I think we would want the same for Unix. Presumably Unix should be added to the `SupportedPlatform` item when the platform is Unix or when it's a platform agnostic build and there is no other Unix specific implementation.

cc @buyaa-n @jeffhandley

Reply:
> [The SDK unconditionally adds supported platforms](https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedPlatforms.props#L16-L18) and one of those is Windows. The SDK itself doesn't make it possible to only target Windows, macOS or Linux via a target platform, but we in dotnet/runtime allow that via the Microsoft.DotNet.TargetFramework.Sdk.

SDK supported platforms cause warnings only for cross platform targets, has no effect targeted builds. I might missing something but  still do not understand why we could not use Linux or Unix targets for the above mentioned build and keep the $(NetCoreAppCurrent) only for real cross platform targets.